dds/DCPS/SafetyProfileStreams.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * $Id$
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+#include "DCPS/DdsDcps_pch.h" //Only the _pch include should start with DCPS/
+
+#include "dds/DCPS/SafetyProfileStreams.h"
+
+namespace OpenDDS {
+namespace DCPS {
+
+OPENDDS_STRING
+to_dds_string(unsigned short int to_convert)
+{
+  const char* fmt = "%hu";
+  int buff_size = size_buffer(fmt, to_convert);
+  char buf[buff_size];
```

Comment:
Arrays need to have compile-time known size.  Use a constant for worst-case # of chars needed for a number.